scipy/spatial/_qhull.pyx

```diff
@@ -2929,6 +2929,20 @@ class HalfspaceIntersection(_QhullUser):
         of halfspaces is also not possible after `close` has been called.
 
         """
+        # We check for non-feasibility of incremental additions
+        # in a manner similar to `qh_sethalfspace`
+        halfspaces = np.atleast_2d(halfspaces)
+        dists = np.dot(halfspaces[:, :self.ndim], self.interior_point) + halfspaces[:, -1]
```

Comment:
This is now vectorized in a manner similar to that described by @RichardBowenGM, and I've added a 3d test as well, just to be cautious.